### ISSUE TITLE
Missing attribute exception in nnet3/train_dnn.py

### ISSUE THREAD

**Issue by pstanisl** [kaldi-asr/kaldi#2352]
Checking `transform_dir` in `train_dnn.py` causing crash

```bash
2018-04-12 10:16:19,291 [steps/nnet3/train_dnn.py:35 - <module> - INFO ] Starting DNN trainer (train_dnn.py)
steps/nnet3/train_dnn.py --stage=-10 --cmd=run.pl --feat.cmvn-opts=--norm-means=false --norm-vars=false --trainer.srand=0 --trainer.max-param-change=2.0 --trainer.num-epochs=3 --trainer.samples-per-iter=400000 --trainer.optimization.num-jobs-initial=2 --trainer.optimization.num-jobs-final=2 --trainer.optimization.initial-effective-lrate=0.0015 --trainer.optimization.final-effective-lrate=0.00015 --trainer.optimization.minibatch-size=256,128 --trainer.num-jobs-compute-prior 2 --egs.dir= --egs.opts --samples-per-iter 400000 --cleanup.remove-egs=true --use-gpu=true --feat-dir=data/100/train --ali-dir=models/100/gmm/tri_4096-lda_mllt-ali --lang data/100/train/lang --reporting.email= --dir=models/100/tdnn
['steps/nnet3/train_dnn.py', '--stage=-10', '--cmd=run.pl', '--feat.cmvn-opts=--norm-means=false --norm-vars=false', '--trainer.srand=0', '--trainer.max-param-change=2.0', '--trainer.num-epochs=3', '--trainer.samples-per-iter=400000', '--trainer.optimization.num-jobs-initial=2', '--trainer.optimization.num-jobs-final=2', '--trainer.optimization.initial-effective-lrate=0.0015', '--trainer.optimization.final-effective-lrate=0.00015', '--trainer.optimization.minibatch-size=256,128', '--trainer.num-jobs-compute-prior', '2', '--egs.dir=', '--egs.opts', '--samples-per-iter 400000', '--cleanup.remove-egs=true', '--use-gpu=true', '--feat-dir=data/100/train', '--ali-dir=models/100/gmm/tri_4096-lda_mllt-ali', '--lang', 'data/100/train/lang', '--reporting.email=', '--dir=models/100/tdnn']
Traceback (most recent call last):
  File "steps/nnet3/train_dnn.py", line 457, in <module>
    main()
  File "steps/nnet3/train_dnn.py", line 440, in main
    [args, run_opts] = get_args()
  File "steps/nnet3/train_dnn.py", line 95, in get_args
    [args, run_opts] = process_args(args)
  File "steps/nnet3/train_dnn.py", line 116, in process_args
    if args.transform_dir is None:
AttributeError: 'Namespace' object has no attribute 'transform_dir'
# Accounting: time=7 threads=1
```

It is possible caused by removing the argument from `egs/wsj/s5/steps/libs/nnet3/train/common.py` in commit https://github.com/kaldi-asr/kaldi/commit/5294666e88755dbdbcff116f65febad8962949f0

Line of code: https://github.com/kaldi-asr/kaldi/blob/ad48342ae826c1196e806a2a6ac483f84523f8df/egs/wsj/s5/steps/nnet3/train_dnn.py#L116-L117

**Comment by pstanisl** [comment]
It is also necessary to delete the argument `transform_dir=args.transform_dir,` in `train_lib.acoustic_model.generate_egs` in `train_dnn.py`.

Line of code:
https://github.com/kaldi-asr/kaldi/blob/ad48342ae826c1196e806a2a6ac483f84523f8df/egs/wsj/s5/steps/nnet3/train_dnn.py#L237

Whole calling:
https://github.com/kaldi-asr/kaldi/blob/ad48342ae826c1196e806a2a6ac483f84523f8df/egs/wsj/s5/steps/nnet3/train_dnn.py#L227-L238

**Comment by danpovey** [comment]
Oops, looks like I missed removing those when I removed that from the scripts    Fixed in #2353.  Thanks.
